lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -879,19 +879,28 @@ function dosomething_campaign_form_campaign_node_form_after_build($form, &$form_
  * @return array
  */
 function dosomething_campaign_get_default_gallery_image_urls($size = '300x300') {
-  $urls = array();
   // Variable name which holds a comma separated list of default File fid's.
   $var_name = 'dosomething_campaign_default_gallery_fids';
-  // Extract the fid's.
-  $fids = explode(',', variable_get($var_name, ''));
-  // Loop through each one:
-  foreach ($fids as $fid) {
-    // Make sure an image exists:
-    if ($url = dosomething_image_get_themed_image_url_by_fid((int) $fid, $size)) {
-      $urls[] = $url;
+  $fids = variable_get($var_name);
+
+  if (is_null($fids)) {
+    return false;
+  }
+  else {
+    $urls = array();
+
+    // Extract the fid's.
+    $fids = explode(',', $fids);
+
+    // Loop through each one:
+    foreach ($fids as $fid) {
+      // Make sure an image exists:
+      if ($url = dosomething_image_get_themed_image_url_by_fid((int) $fid, $size)) {
+        $urls[] = $url;
+      }
     }
+    return $urls;
   }
-  return $urls;
```

Comment:
The return `FALSE` happens but only if the `$fids` variable is `NULL`. There's a conditional there to decide that.